ops/lib.rs

```diff
@@ -91,7 +92,15 @@ pub fn op(_attr: TokenStream, item: TokenStream) -> TokenStream {
 
 /// Generate the body of a v8 func for an async op
 fn codegen_v8_async(core: &TokenStream2, f: &syn::ItemFn) -> TokenStream2 {
-  let (arg_decls, args_tail) = codegen_args(core, f, 1, 2);
+  let arg0 = f.sig.inputs.first();
+  let uses_opstate = arg0.map(is_rc_refcell_opstate).unwrap_or_default();
+  let args_head = if uses_opstate {
+    quote! { state, }
+  } else {
+    quote! {}
+  };
+  let rust_i0 = if uses_opstate { 1 } else { 0 };
+  let (arg_decls, args_tail) = codegen_args(core, f, rust_i0, 2);
```

Comment:
```suggestion
  let (arg_decls, args_tail) = codegen_args(core, f, usize::from(uses_opstate), 2);
```

Reply:
I think coercing bools to arg offsets could hinder maintainability more than the verbosity of the if expr